torch/utils/_python_dispatch.py

```diff
@@ -51,6 +53,20 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         _pop_mode()
 
+    def check_mode_state_push(self):
```

Comment:
bikeshedding on these names welcome

Idea of this is they're the functions that checks that the mode isn't on the stack already and updates its state to reflect a push or pop